extension/src/paymentHandler/payment-utils.js

```diff
@@ -190,6 +190,13 @@ function isValidMetadata(str) {
   return str.indexOf(' ') < 0
 }
 
+function getIdempotencyKey(transaction) {
+  let idempotencyKey = transaction.custom?.fields?.idempotencyKey
```

Comment:
where do you initialize `transaction.custom.fields.idempotencyKey`? I cannot find it in you PR

Reply:
I saw [this part](https://github.com/commercetools/commercetools-adyen-integration/blob/d349060ef9a8eafac34429f981904d338b5b95bf/extension/src/service/web-component-service.js#L65) and it is clear for me now. I thought the program would put the generated `idempotencyKey` back to transasction custom field at some points

Reply:
I see. Let me check if this could be confusing for the user or not.

Reply:
@leungkinghin https://github.com/commercetools/commercetools-adyen-integration/pull/1010/commits/1aadf8b15b7f2a8e3c07de939600a4fa275995af